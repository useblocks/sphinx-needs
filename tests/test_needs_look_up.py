import json
from pathlib import Path

import pytest
from syrupy.filters import props


@pytest.mark.parametrize(
    "test_app", [{"buildername": "needs_lut", "srcdir": "doc_test/doc_needs_builder"}], indirect=True
)
def test_doc_needs_lut_builder(test_app, snapshot):
    app = test_app
    app.build()
    needs_list = json.loads(Path(app.outdir, "needs_lut.json").read_text())
    assert needs_list == snapshot(exclude=props("created"))
