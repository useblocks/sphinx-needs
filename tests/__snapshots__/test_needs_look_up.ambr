# serializer version: 1
# name: test_doc_needs_lut_builder[test_app0]
  dict({
    'current_version': '1.0',
    'project': 'Python',
    'versions': dict({
      '1.0': dict({
        'filters': dict({
        }),
        'filters_amount': 0,
        'needs': dict({
          'TC_001': 'index',
          'TC_NEG_001': 'index',
          'US_63252': 'index',
        }),
        'needs_amount': 3,
      }),
    }),
  })
# ---
